tests/ttnn/unit_tests/operations/test_maxpool2d.py

```diff
@@ -153,6 +153,205 @@ def run_max_pool(
+
+    if in_c / max_cores < 16:
+        pytest.skip("Width sharding requires large enough channels to shard (at least 16 per core)")
+
+    torch.manual_seed(0)
+    torch.set_printoptions(precision=3, sci_mode=False, linewidth=500, threshold=10000, edgeitems=32)
+
+    ## construct the tensor in NCHW shape
+    act = torch.randn(act_shape, dtype=torch.bfloat16)
+    ## this op expects input tensor as { N, 1, H * W, C }, so rearrange and reshape tensor
+    ## but before that, make sure in_c is multiple of tile width
+    act_shape = (1, 1, in_n * in_h * in_w, in_c)
+    act_permuted = torch.permute(act, (0, 2, 3, 1))
+    act_reshaped = act_permuted.reshape(act_shape)
+
+    if dtype == ttnn.bfloat8_b:
+        if (in_h * in_w) % 32 != 0:
+            pytest.skip("For BFP8_B datatype, input height * width should be multiple of 32")
+        if (in_c / max_cores) % 32 != 0:
+            pytest.skip("For BFP8_B datatype, input channels / cores should be multiple of 32")
+        ttact = ttnn.from_torch(act_reshaped, dtype, layout=ttnn.TILE_LAYOUT)
+    else:
+        ttact = ttnn.from_torch(act_reshaped, dtype)
+
+    ttact_device = ttnn.to_device(ttact, device)
+
+    output = ttnn.max_pool2d(
+        input_tensor=ttact_device,
+        batch_size=in_n,
+        input_h=in_h,
+        input_w=in_w,
+        channels=in_c,
+        kernel_size=[kernel_h, kernel_w],
+        stride=[stride_h, stride_w],
+        padding=[pad_h, pad_w],
+        dilation=[dilation_h, dilation_w],
+        memory_config=None,
+        applied_shard_scheme=ttnn.TensorMemoryLayout.WIDTH_SHARDED,
+    )
+
+    output_host = output.cpu()
+    output_pytorch_padded = torch.Tensor(ttnn.to_torch(output_host))
+    output_pytorch = output_pytorch_padded[:, :, :, :in_c]
+
+    ## reference
+    golden_pytorch = torch.nn.MaxPool2d(
+        kernel_size,
+        stride=stride,
+        padding=padding,
+        dilation=dilation,
+        return_indices=False,
+        ceil_mode=False,
+    )(act)
+
+    ## test for equivalance
+    golden_shape = golden_pytorch.shape
+    output_pytorch = output_pytorch.reshape(golden_shape[0], golden_shape[2], golden_shape[3], golden_shape[1])
+
+    output_pytorch = torch.permute(output_pytorch, (0, 3, 1, 2))  ## N, C, H, W
+    passing, pcc = assert_with_pcc(output_pytorch, golden_pytorch)
+
+    logger.debug(f"Passing: {passing}, PCC: {pcc}")
+
+    ## do more rigorous comparision for each element
+    atol, rtol = torch.testing._comparison.default_tolerances(torch.bfloat16)
+    if dtype == ttnn.bfloat8_b:
+        atol = 0.35
+
+    allclose = torch.allclose(output_pytorch, golden_pytorch, atol=atol)
+    isclose = torch.all(torch.isclose(output_pytorch, golden_pytorch, atol=atol))
+    isequal = torch.equal(output_pytorch, golden_pytorch)
+
+    assert allclose
+    assert isclose
+    if dtype == ttnn.bfloat16:
+        assert isequal
+
+
+def run_max_pool_block_shard(
```

Comment:
run_max_pool_block_shard and run_max_pool_width_shard seems very similar can we merge them?

Reply:
will do